subscriptions-api-openapi.yaml

```diff
@@ -135,7 +137,7 @@ components:
           example: "HTTP"
         protocolsettings:
           type: object
-          anyOf:
+          oneOf:
```

Comment:
Are you sure we want this change? As I understand it, this will fail validation if it input matches more than one of the $ref's. But what if that's true? What if the settings for two protocols look the exact same - that seems legal to me. I think "anyOf" might be better since we just want to make sure it matches "at least one", not "exactly one".

Reply:
Yes, good catch, I think you're right. Short answer: the OAS spec isn't very clear about oneOf rendering ambiguous objects invalid, but the implementation of the openapi-generator, for example, matches your description. The issue I am experiencing with the openapi-generator for golang seems to be a separate problem.